shared/constants/chat2/index.tsx

```diff
@@ -1162,9 +1162,9 @@ export const _useState = Z.createZustand<State>((set, get) => {
           get().dispatch.onChatInboxSynced(action)
           break
         case EngineGen.chat1ChatUiChatInboxLayout:
+          get().dispatch.updateInboxLayout(action.payload.params.layout)
```

Comment:
this is a bug that can cause a convo switch due to a stale value being reference in the calls below (maybeChangeSelectedConv etc). we need to update the state first